### PR TITLE
ghc 8.4:  update configuration

### PR DESCRIPTION
**Pull request by deepfire**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by deepfire** [comment]
@peti, this updates the package set in two ways:

1. The updates to `securemem`, `deepseq-generics` and `funcmp` have been interned
2. The trims are in accord with the latest nh: https://github.com/deepfire/nh
3. The override DB is published at: https://github.com/deepfire/configuration-ghc-8.4.x

File: pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix [hunk 1]
```diff
@@ -88,12 +88,34 @@ self: super: {
...
+  ##       from the context: (Monad m, Monoid a)
+  free = super.free_5;
+
+  funcmp = overrideCabal super.funcmp_1_9 (drv: {
+    ## Needs bump to a versioned attribute
+    ## 
+    ## Needed for (<>) in prelude
+    ## Setup: Encountered missing dependencies:
+    ## base >=3 && <4.11
+    jailbreak       = true;
```
Comment (peti) [inline pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix]: I am surprised that jailbreak is necessary. I released funcmp 1.9 specifically to address ghc 8.4.x compatibility. Did I make a mistake somewhere? According to http://hackage.haskell.org/package/funcmp, the 1.9 version accepts ` base >= 4.5 && < 5`.

Comment (deepfire) [reply pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix]: Something must have messed up in the gearbox, investigating..

Comment (deepfire) [reply pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix]: What I ran to facilitate investigation:
```
nh audit --skip-baseline
Generating override cache: 63/63
Assembling overrides................................................................. done.
                  ChasingBottoms: no upstream specified
                      haskell-gi: hackage 0.21.0 obsoletes upstream fix (version 0.21.0), committed Tue Jan  9 19:38:03 MSK 2018 < release uploaded Thu, 11 Jan 2018 09:57:02 GMT
                 haskell-gi-base: hackage 0.21.0 obsoletes upstream fix (version 0.21.0), committed Tue Jan  9 19:38:03 MSK 2018 < release uploaded Thu, 11 Jan 2018 09:55:04 GMT
               vector-algorithms: no upstream specified
deepfire@andromedae:~/holotype$ nh trim --skip-baseline funcmp
Generating override cache: 63/63
-( trimming in progress
TRIM funcmp>
TRIM funcmp -all>  Assembling overrides................................................................. done.
FATAL: failed to build 'funcmp', without overriding 'default'
FATAL: logs:  nh log funcmp default
CABAL-MISSING-DEPS
Setup: Encountered missing dependencies:
base >=3 && <4.11
FAILED
TRIM funcmp -jailbreak>    ## Needs bump to a versioned attribute
  ##
  ## Needed for (<>) in prelude  funcmp = super.funcmp_1_9;

Assembling overrides................................................................. done.
FATAL: failed to build 'funcmp', without overriding 'jailbreak'
FATAL: logs:  nh log funcmp jailbreak
CABAL-MISSING-DEPS
Setup: Encountered missing dependencies:
base >=3 && <4.11
FAILED
TRIM funcmp -src>  SKIPPED, explanation exists:  nh x emit_explanation src funcmp
deepfire@andromedae:~/holotype$ nh info funcmp
attribute 'funcmp':

           status: shadowed
        overrides: jailbreak src
  nixpkgs release: 1.8
 upstream release: 1.9
 override release: 1.9
             meta: src.explanation
```
What I see from the above, is that https://raw.githubusercontent.com/NixOS/nixpkgs/master/pkgs/development/haskell-modules/hackage-packages.nix still defines `funcmp` to be 1.8 -- which still has the problem for any default users of `funcmp` on GHC 8.4 -- even if they wouldn't have it if they used `funcmp_1_9`.

Comment (deepfire) [reply pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix]: cc @peti

Comment (deepfire) [reply pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix]: The `without overriding 'default'` "explanation" is sufficiently confusing it needs attention, I guess..  It actually means "without overriding _anything_" or "by _default_"..

Comment (peti) [reply pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix]: Well, `funcmp` does not compile on GHC 8.4.x even after the jailbreak `funcmp_1_9` compiles fine on GHC 8.4.x without the jailbreak. Adding the jailbreak to the 1.9 version does not hurt, but it seems unnecessary to me (and it's not going to benefit users of the 1.8 version in any way, either).

Comment (deepfire) [reply pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix]: You're right, the trimmer has messed up and was building `funcmp-1.8` while attempting to trim the jailbreak override -- that's clearly a bug..

**Pull request merged**